src/Microsoft.AspNet.Mvc.DataAnnotations/AttributeAdapterBase.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNet.Mvc.DataAnnotations
+{
+    /// <summary>
+    /// An extension of <see cref="ValidationAttributeAdapter{TAttribute}"/> which forces adapters to implement
+    /// GetErrorMessage while providing their relevent values.
```

Comment:
Having trouble understanding what this comment is trying to say. Why is it more than `An abstract subclass of <see cref="ValidationAttributeAdapter{TAttribute}"/> which implements <see cref="IAttributeAdapter"/>.`?